### PR TITLE
Alma Availability

### PR DESCRIPTION
**Pull request by tpendragon**
Starts implementing Alma availability endpoint for a single :id.

Work towards #573

### CODE REVIEW

File: spec/controllers/availability_controller_spec.rb [hunk 1]
```diff
@@ -27,6 +27,36 @@
...
+          "location" => "sa",
+          # "copy_number" => 1, TODO => Implement
+          "item_id" => "2382456270006421"
+          # "on_reserve": "N", TODO: Implement
+          # "patron_group_charged": null, TODO: Implement
+          # "status": "On-Site", TODO: Implement
+          # "label": "Marquand Library", TODO: Implement
+          # "status_label": "On-site access" TODO: Implement
+        },
+        # This isn't in Voyager's bibdata, I don't know why.
```
Comment (tpendragon) [inline spec/controllers/availability_controller_spec.rb:52]: @mzelesky Do you know why 991227850000541 has a holding in Alma but not from production bibdata? Is there something about this holding that makes it so it shouldn't show up, or is the data just different?

File: app/adapters/alma_adapter/availability_status.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
+class AlmaAdapter
+  class AvailabilityStatus
+    def self.from_bib(bib:)
```
Comment (hackartisan) [inline app/adapters/alma_adapter/availability_status.rb:3]: Can we document the param here?

**Review by hackartisan — CHANGES_REQUESTED** [review]
This is a great start here thank you!! Just one small request.

File: spec/controllers/availability_controller_spec.rb (hunk 1 above)
Comment (christinach) [reply spec/controllers/availability_controller_spec.rb:52]: I think we added an item while we were testing.

**Review by christinach — CHANGES_REQUESTED** [review]

File: app/adapters/alma_adapter/availability_status.rb (hunk 2 above)
Comment (tpendragon) [reply app/adapters/alma_adapter/availability_status.rb:3]: Sure. I wasn't sure how stable this api would be, but can always update it.

File: app/adapters/alma_adapter/availability_status.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      holdings.each_with_object({}) do |holding, acc|
+        acc[holding["holding_id"]] = holding_summary(holding)
+      end
+    end
+
+    def holding_summary(holding)
+      {
+        more_items: holding["total_items"].to_i > 1,
+        location: marc_record["852"]["b"], # Location of bib record
+        item_id: item_data[holding["holding_id"]]&.first&.item_data&.fetch("pid", nil)
```
[the remark this answers is not in the document]
Comment (christinach) [reply app/adapters/alma_adapter/availability_status.rb:22]: The location code is 852 | c. The 852 | b that yo include in the code is the location name.
example:
`852 0  $b MAIN $c main $h HC1000 $i .O48 2001 $8 2281654580006421 `

[the remark this answers is not in the document]
Comment (christinach) [reply app/adapters/alma_adapter/availability_status.rb:22]: The **location code** is 852 | c. The 852 | b that you include in the code is the **library cod**e.
example:
`852 0  $b MAIN $c main $h HC1000 $i .O48 2001 $8 2281654580006421 `

[the remark this answers is not in the document]
Comment (mzelesky) [reply app/adapters/alma_adapter/availability_status.rb:22]: A bib record does not have a location associated with it. Could you explain where you're seeing this?

[the remark this answers is not in the document]
Comment (tpendragon) [reply app/adapters/alma_adapter/availability_status.rb:22]: I get it from the bibliographic endpoint.

[the remark this answers is not in the document]
Comment (mzelesky) [reply app/adapters/alma_adapter/availability_status.rb:22]: If there is an 852 in the bibliographic endpoint, it is from the Voyager bib record. When I exported records from Voyager to input them into Alma, I retained 852 fields to handle mapping. The bib API for Alma does not put 852 fields in the record for holdings info. Holding info is put in AVA fields.

Any 852 field returned from the bib API in Alma should be disregarded.

[the remark this answers is not in the document]
Comment (tpendragon) [reply app/adapters/alma_adapter/availability_status.rb:22]: Okay, I'll take this out of this test and add a checkbox to #573

**Review by hackartisan — APPROVED** [review]

File: spec/controllers/availability_controller_spec.rb (hunk 1 above)
Comment (mzelesky) [reply spec/controllers/availability_controller_spec.rb:52]: Yes, as @christinach said, the MAIN-main item was added in Alma during testing.

**Review by christinach — APPROVED** [review]

**Pull request merged**